numba_rvsdg/core/datastructures/block_map.py

```diff
@@ -318,3 +320,56 @@ def join_tails_and_exits(self, tails: Set[Label], exits: Set[Label]):
     @staticmethod
     def bcmap_from_bytecode(bc: dis.Bytecode):
         return {inst.offset: inst for inst in bc}
+
+    @staticmethod
+    def from_yaml(yaml_string):
+        data = yaml.safe_load(yaml_string)
+        return BlockMap.from_dict(data)
+
+    @staticmethod
+    def from_dict(graph_dict):        
+        block_map_graph = {}
+        clg = ControlLabelGenerator()
+        for index, attributes in graph_dict.items():
+            jump_targets = attributes["jt"]
+            backedges = attributes.get("be", ())
+            label = ControlLabel(str(clg.new_index()))
+            block = BasicBlock(
+                label=label,
+                backedges=wrap_id(backedges),
+                _jump_targets=wrap_id(jump_targets),
+            )
+            block_map_graph[label] = block
+        return BlockMap(block_map_graph, clg=clg)
+
+    def to_yaml(self):
+        # Convert to yaml
+        block_map_graph = self.graph
+        yaml_string = """"""
+
+        for key, value in block_map_graph.items():
+            jump_targets = [f"{i.index}" for i in value._jump_targets]
+            jump_targets = str(jump_targets).replace("\'", "\"")
+            back_edges = [f"{i.index}" for i in value.backedges]
+            yaml_string += f"""
+    "{str(key.index)}":
+        jt: {jump_targets}"""         
+            if back_edges:
+                back_edges = str(back_edges).replace("\'", "\"")
+                yaml_string += f"""
+        be: {back_edges}"""
+        return yaml_string
+
+    def to_dict(self):
+        block_map_graph = self.graph
+        graph_dict = {}
+        for key, value in block_map_graph.items():
+            curr_dict = {}
+            curr_dict["jt"] = [f"{i.index}" for i in value._jump_targets]
+            if value.backedges:
+                curr_dict["be"] = [f"{i.index}" for i in value.backedges]
+            graph_dict[str(key.index)] = curr_dict
+        return graph_dict
+
+def wrap_id(indices: Set[Label]):
```

Comment:
Do you think this could go into `test_utils.py` too?

Reply:
No, it's being used by functions within `block_map.py`. 

Besides, it'll go away anyways while the planned library rewrites so we need not bother anyway.